test/integration/hds_integration_test.cc

```diff
@@ -179,8 +179,9 @@ class HdsIntegrationTest : public HttpIntegrationTest,
     if (address->ip()->port() != endpoint.endpoint().address().socket_address().port_value()) {
       return false;
     }
+    // SocketAddress.address is a proto string, not std::string, so convert it before comparing.
     if (address->ip()->addressAsString() !=
-        endpoint.endpoint().address().socket_address().address()) {
+        std::string(endpoint.endpoint().address().socket_address().address())) {
```

Comment:
You could also wrap both sides in absl::string_view, which would not involve any allocation or buffer-copies.

Reply:
actually you can wrap just one side, as absl::string_view can compare itself with strings.